lib/xenopsd.ml

```diff
@@ -71,6 +71,16 @@ let vm_guest_agent_xenstore_quota = ref 128
 
 let vm_guest_agent_xenstore_quota_warn_interval = ref 3600
 
+(* for backward compatibility compute how much memory N entries
+   would've taken *)
+let max_bytes_of_entries entries =
+  (* defaults from oxenstored.conf *)
+  let default_path_max = 1024 in
+  let default_maxsize = 2048 in
```

Comment:
Would it then not be appropriate to read these values from `oxenstored.conf` rather than hardcoding values which could be changed?

As an aside it's interesting that the ocaml version doesn't respect the wire protocol as defined in http://xenbits.xen.org/gitweb/?p=xen.git;a=blob;f=xen/include/public/io/xs_wire.h;h=4dd6632669619a5137fe2069d881b12c399ccc6f;hb=HEAD#l133

Reply:
Regarding path-max we use a lower value than upstream default (it is configurable in oxenstored.conf, the upstream default matches C xenstored). Given all the recent security issues around OOM we think it is prudent to keep it at 1024.

Agreed on parsing oxenstored.conf, that could be done in a new commit on top of this.

Reply:
Added oxenstored.conf parsing